engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java

```diff
@@ -828,7 +828,12 @@ private static YarnConfiguration getYarnConfiguration(Configuration conf)
 
   public static InetSocketAddress getRMWebAddress(Configuration conf, String rmId)
   {
-    boolean sslEnabled = conf.getBoolean(CommonConfigurationKeysPublic.HADOOP_SSL_ENABLED_KEY, CommonConfigurationKeysPublic.HADOOP_SSL_ENABLED_DEFAULT);
+    /*
+     * This is far from confirmed as the "correct" fix.  The previous constants HADOOP_SSL_ENABLED_DEFAULT and HADOOP_SSL_ENABLED_KEY have
+     * been deprecated as far back as I can find Hadoop versions (2.4.1 at least), HADOOP_CALLER_CONTEXT_ENABLED_DEFAULT and
+     * HADOOP_CALLER_CONTEXT_ENABLED_KEY are the only two constants in 3.1.0 that seem even remotely related...
+     * */
+    boolean sslEnabled = conf.getBoolean(CommonConfigurationKeysPublic.HADOOP_CALLER_CONTEXT_ENABLED_KEY, CommonConfigurationKeysPublic.HADOOP_CALLER_CONTEXT_ENABLED_DEFAULT);
```

Comment:
My comments on this:

- instead of having a tentative fix like this why can't we have a permanent fix? See below
- Do we know what string constants these identifiers CommonConfigurationKeysPublic.HADOOP_SSL_ENABLED_KEY and CommonConfigurationKeysPublic.HADOOP_CALLER_CONTEXT_ENABLED_KEY stand for?
- Same thing for CommonConfigurationKeysPublic.HADOOP_SSL_ENABLED_DEFAULT and CommonConfigurationKeysPublic.HADOOP_CALLER_CONTEXT_ENABLED_DEFAULT
- if they are the same (which I doubt) we haven't introduced any incompatibility as far as external user impact is concerned.
- if they are different we can just break the compatibility (not desirable) by using these new property names
- or (desirable) honor the old property value by defining it here and checking for it before checking the new value.\
- the comment is also repeated elsewhere. Ideally it should only exist in one place.

Reply:
You make a good point...wil look into the constants more...the weird thing is that in the Javadocs for Hadoop, those two constants have no information about them as far back as I can see...other than being marked as deprecated...will do some more digging.  From the APEX side of the house, what are we doing with this function other than the obvious...checking to see if SSL is enabled.

Reply:
I don't know how reliable this page is...but the string/boolean values for the constants seem to make sense...I will hunt down which constants, if any, replaced them...

```java

public static final String HADOOP_SSL_ENABLED_KEY = "hadoop.ssl.enabled"
public static final boolean HADOOP_SSL_ENABLED_DEFAULT = false
```

Reply:
It might be worthwhile asking on the users@ list if anyone is using Apex in SSL mode. If no-one is using SSL we don't need to worry about breaking compatibility. But we should still confirm that HADOOP_CALLER_CONTEXT_ENABLED_KEY is the right property to check to verify SSL is enabled in the user's Hadoop cluster.

Reply:
Oh, I think our comments crossed. And what's the value of `HADOOP_CALLER_CONTEXT_ENABLED_KEY` ?

Reply:
@vrozov who is recommending dropping SSL support? I didn't suggest that if that's the impression you got. My question was in connection with creating incompatibility which might be okay if no-one is currently using SSL